code/mob/living/intangible/flock/flockmind.dm

```diff
@@ -28,10 +30,33 @@
 	if (!F)
 		src.addAbility(/datum/targetable/flockmindAbility/spawnEgg)
 		src.addAbility(/datum/targetable/flockmindAbility/ping)
+		src.addAbility(/datum/targetable/flockmindAbility/tutorial)
 	else
 		src.started = TRUE
 		src.addAllAbilities()
 
+
+/mob/living/intangible/flock/flockmind/proc/start_tutorial()
+	if (src.tutorial)
+		return
+	src.tutorial = new(src)
+	if (src.tutorial.initial_turf)
+		src.tutorial.Start()
+	else
+		boutput(src, "<span class='alert'>Could not start tutorial! Please try again later or call Wire.</span>")
+		src.tutorial = null
+
+/mob/living/intangible/flock/flockmind/select_drone(var/mob/living/critter/flock/drone/drone)
+	if(src.tutorial && !src.tutorial.PerformAction(FLOCK_ACTION_DRONE_SELECT))
+		return
+	..()
+
+/mob/living/intangible/flock/flockmind/Move(NewLoc, direct)
+	if (isturf(NewLoc))
+		if (istype(NewLoc, /turf/cordon))
+			return FALSE
+	..()
```

Comment:
Since intangible doesn't check `Cross` or anything I don't think it's possible to do the check on the turf itself, listening post walls also don't block intangibles anymore.

Reply:
Yeah whatever works, just seems off checking for cordon turf when majority of turf isn't that I guess